global-api/importer/carbon_monitor/process.py

```diff
@@ -0,0 +1,150 @@
+import concurrent.futures
+import os
+import uuid
+
+import pandas as pd
+import openclimate
+
+client = openclimate.Client()
+
+def get_parts_of_iso(client, iso):
+    df_parts = client.parts(actor_id=iso, part_type="adm1").get("actor_id")
+    return list(df_parts) + [iso]
+
+
+def get_cities_from_part(client, part):
+    try:
+        df_tmp = client.parts(actor_id=part, part_type="city").loc[:, ["actor_id", "name"]]
+        df_tmp["is_part_of"] = part
+        return df_tmp
+    except:
+        pass
+
+def uuid_generate_v3(name, namespace=uuid.NAMESPACE_OID):
+    """generate a version 3 UUID from namespace and name"""
+    assert isinstance(name, str), "name needs to be a string"
+    assert isinstance(namespace, uuid.UUID), "namespace needs to be a uuid.UUID"
+    return str(uuid.uuid3(namespace, name))
+
+
+if __name__ == "__main__":
+    INPUT_FILE = './raw/carbon-monitor-cities-all-cities-FUA-v0325.csv'
+    INPUT_FILE = os.path.abspath(INPUT_FILE)
+
+    OUTPUT_FILE = './processed/carbon-monitor-cities-all-cities-FUA-v0325_processed.csv'
+    OUTPUT_FILE = os.path.abspath(OUTPUT_FILE)
+
+    KT_TO_KG = 1_000_000
+
+    COUNTRY_REPLACE_DICT = {
+        "country": {
+            "United States": "United States of America",
+            "United Kingdom": "United Kingdom of Great Britain and Northern Ireland",
+            "Vietnam": "Viet Nam",
+            "Korea": "Korea, the Republic of",
+            "Turkey": "Türkiye",
+            "Russia": "Russian Federation",
+        }
+    }
+
+    df = (
+        pd.read_csv(INPUT_FILE, parse_dates=["date"])
+        # 0. filter out rows with 0 emissions
+        .loc[lambda x: x['value (KtCO2 per day)']>0]
```

Comment:
Why are you filtering out the 0 values? Wouldn't these still be valid and good to have present, so the user doesn't have to research them themselves?

Or are these invalid

Reply:
I discussed with @lgloege and our consensus is that it probably makes sense to be conservative in what we import. Zeroes may be measured values, but they may also signal a lack of information.